src/pages/build-apps/guides/data-storage.md

```diff
@@ -0,0 +1,213 @@
+
+```jsx
+const options = {
+  user: 'ryan.id', // the Stacks ID of the user for which to lookup the file
+  app: 'https://BlockstackApp.com', // origin of the app this file is stored for
+  decrypt: false,
+};
+
+const userSession = new UserSession();
+storage.getFile('hello.txt', options).then(fileContents => {
+  // get the contents of the file /message.txt
+  assert(fileContents === 'hello world!');
+});
+```
+
+## Delete a file
+
+Use the [`UserSession.deleteFile`](https://blockstack.github.io/stacks.js/classes/storage.html#deletefile) from the application's data store.
+
+```jsx
+const userSession = new UserSession();
+const storage = new Storage({ userSession });
+
+storage.deleteFile('hello.txt').then(() => {
+  // hello.txt is now removed.
+});
+```
+
+## Write-to and Read-from URL Guarantees
+
+Gaia is built on a driver model that supports many storage services. So, with
+very few lines of code, you can interact with providers on Amazon S3, Dropbox,
+and so forth. The simple `getFile()` and `putFile()` interfaces are kept simple
+because Stacks assumes and wants to encourage a community of
+open-source-data-management libraries.
+
+The performance and simplicity-oriented guarantee of the Gaia specification is
+that when an application submits a write-to
+`https://myhub.service.org/store/foo/bar` URL, the application is guaranteed to
+be able to read from the `https://myreads.com/foo/bar` URL. Note that, while the
+prefix in the write-to url (for example,`myhub.service.org/store`) and the read-from URL
+(`https://myreads.com`) are different, the `foo/bar` suffixes are the same.
+
+By default, `putFile()` encrypts information while `getFile()` decrypts it by default. Data stored in an
+encrypted format means only the user that stored it can view it. For applications that want other users to
+view data, the application should set the `encrypt` option to `false`. And, corresponding, the `decrypt`
+option on `getFile()` should also be `false`.
+
+Consistent, identical suffixes allow an application to know _exactly_ where a
+written file can be read from, given the read prefix. The Gaia service defines a `hub_info` endpoint to obtain
+that read prefix:
+
+```bash
+GET /hub_info/
+```
+
+The endpoint returns a JSON object with a `read_url_prefix`, for example, if my service returns:
+
+```jsx
+{ ...,
+  "read_url_prefix": "https://myservice.org/read/"
+}
+```
+
+The data be read with this `getFile()` and this address:
+
+```
+https://myservice.org/read/1DHvWDj834zPAkwMhpXdYbCYh4PomwQfzz/0/profile.json
+```
+
+The application is guaranteed that the profile is written with `putFile()` this request address:
+
+```
+https://myservice.org/store/1DHvWDj834zPAkwMhpXdYbCYh4PomwQfzz/0/profile.json
+```
+
+When you use the `putFile()` method it takes the user data and POSTs it to the user's Gaia storage hub.
+The data POSTs directly to the hub, the blockchain is not used and no data is stored there. The limit on
+file upload is currently 25mb.
```

Comment:
Hm, this makes me think that we should add a note about the Hiro-hosted hub, since there technically that limit is configurable - but for all practical purposes, the limit is 25MB. Also the correct unit for "megabytes" is MB (capitalized).